File: backend spring/src/main/java/com/bloodbank/entity/DonorProfile.java
package com.bloodbank.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;
//import java.util.Date;

@Entity
@Table(name = "donor_profiles")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class DonorProfile {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "profile_id")
    private Long profileId;

    @OneToOne
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    private User user;

    @Column(name = "full_name", nullable = false)
    private String fullName;

    @Column(name = "blood_group", nullable = false, length = 5)
    private String bloodGroup;

    @Column(name = "contact_number", nullable = false, length = 15)
    private String contactNumber;

//    @Temporal(TemporalType.DATE)
//    @Column(name = "last_donation_date")
//    private Date lastDonationDate;

 // TO THIS (The @Temporal annotation is no longer needed):
    @Column(name = "last_donation_date")
    private LocalDate lastDonationDate;
    
    // Extra Columns from our DB design
    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;
    
    @Column(name = "is_eligible", nullable = false)
    private boolean isEligible = true;
}

File: BackendDotNet/Models/DonorProfile.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace BackendDotNet.Models
{
    [Table("donor_profiles")]
    public class DonorProfile
    {
        [Key]
        [Column("profile_id")]
        public long ProfileId { get; set; }

        [Required]
        [Column("user_id")]
        public long UserId { get; set; }

        [Required]
        [Column("full_name")]
        public string FullName { get; set; } = string.Empty;

        [Required]
        [Column("blood_group")]
        [StringLength(5)]
        public string BloodGroup { get; set; } = string.Empty;

        [Required]
        [Column("contact_number")]
        [StringLength(15)]
        public string ContactNumber { get; set; } = string.Empty;

        [Column("last_donation_date")]
        public DateTime? LastDonationDate { get; set; }

        [Column("age")]
        public int Age { get; set; }

        [Column("gender")]
        [StringLength(10)]
        public string Gender { get; set; } = string.Empty;

        [Column("address")]
        public string Address { get; set; } = string.Empty;

        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("updated_at")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        [Column("is_eligible")]
        public bool IsEligible { get; set; } = true;

        // Navigation property
        [ForeignKey("UserId")]
        [JsonIgnore]
        public virtual User User { get; set; } = null!;
    }
}
